ARCHITECTURE.md

```diff
@@ -0,0 +1,265 @@
+protocol (often abbreviated OIDC). OIDC is an industry-wide standard protocol based on the
+[OAuth2.0](https://tools.ietf.org/html/rfc6749) framework. In order to understand
+how our libraries work internally, some understanding of OAuth/OIDC is definitely preferable.
+
+### A short glossary
+
+Here is a list of terms having a specific meaning in the context of OIDC:
+- **Resource Owner**: the user, who owns resources, some of which are private.
+The notion of Resource Owner in the OIDC sense is broader than the notion of Pod Owner
+in Solid: anyone with a WebID is a Resource Owner, in the authentication sense.
+Whether a particular Resource Owner is **_authorized_** to access a particular Resource
+is a different matter, up to the Solid authorization system (e.g., ACP or ACL), which is
+outside the scope of this document.
+- **Resource Server**: the server hosting resources owned by the Resource Owner.
+In our case, a Solid server. A Resource Server receives requests authenticated
+with an Access Token. Example: https://pod.inrupt.com. 
+- **OIDC issuer**: the Solid Identity Provider, which issues Access Tokens, ID
+tokens, and Refresh Tokens. These tokens tell the Resource Server that the user
+has control over a certain identity (WebID), which can then use that information
+to decide whether to give or deny access. Example: https://broker.pod.inrupt.com.
+- **Client**: the application the Resource Owner uses to access resources on
+a Resource Server. Those resources may be owned directly by the Resource Owner
+(i.e., resources in a Pod created by the resource owner), or on a Resource Server
+hosting someone else's resources, but to which this client application and user
+have been explicitly granted access).
+ Technically, OAuth is a delegation protocol: the Resource Owner allows the
+Client to interact with a Resource Server on its behalf. Example: 
+https://podbrowser.inrupt.com.
+
+### Solid-OIDC
+
+With the [Solid-OIDC specification](https://solid.github.io/authentication-panel/solid-oidc/), 
+Solid extends the OIDC protocol in order to allow it fit better into a decentralized ecosystem,
+specifically with Client WebIDs and DPoP tokens:
+
+- Solid-OIDC introduces the notion of Client WebID, which enables
+Client-managed identifiers instead of Issuer-managed identifiers. By using identifiers
+they control, Clients are no longer required to get their identifiers from the Issuer
+through either static or dynamic client registration.
+
+- Solid-OIDC also makes the support for [Key-bound Access Tokens](https://tools.ietf.org/html/draft-fett-oauth-dpop-04)
+(referred to as DPoP tokens) mandatory: it is only optional in traditional OIDC, where
+Bearer tokens are the default option. DPoP tokens cannot be replayed by a Resource
+Server to another Resource Server, which is an important security feature in a
+decentralized ecosystem such as Solid's.
+
+### Mapping OIDC flows to the code
+
+#### Unsupported flows
```

Comment:
Should this section follow the auth code flow and the refresh code flow? so that we hit the supported flows first?